### PR TITLE
treewide: http -> https sources

### PR DESCRIPTION
**Pull request by infinisil**
This updates the source urls of all top-level packages from http to https where possible. This is done using https://github.com/Infinisil/nix-url-verify which is a mess right now, but the basic functionality works. While it will be able to change urls in regards to redirects too, I think http -> https is a good start. This is in the sense of the recent #42665 by @ryantm

What my program does is:
- Go through all src urls of top-level attributes
- If they are using http, make a request to the https version
- If the request succeeds (2xx status code), the url gets updated

This does not verify the hashes, which shouldn't be needed anyways. Because the hashes stay the same, this also shouldn't trigger any builds.

Here is a list of all attributes whose source url has been changed: https://gist.github.com/d64724f7c284ce8dd4fed9bb0748dde6

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/tools/misc/buildtorrent/default.nix [hunk 1]
```diff
@@ -6,7 +6,7 @@ stdenv.mkDerivation rec {
 	name = "buildtorrent-${version}";
 
 	src = fetchurl {
-		url = "http://mathr.co.uk/blog/code/${name}.tar.gz";
+		url = "http://mathr.https://mathr.co.uk/blog/code/z";.tar.gz
```
Comment (Infinisil) [inline pkgs/tools/misc/buildtorrent/default.nix]: It somehow messed up here.. Not sure what happened

**Comment by Infinisil** [comment]
I figured it out, that file used tabs instead of spaces, which apparently can trip up hnix, ping @jwiegley 

@GrahamcOfBorg eval

**Comment by xeji** [comment]
Thanks! 
LGTM. Please fix the conflicts so we can merge this.

**Comment by grahamc** [comment]
@LnL7 asks a great question -- can the bootstrap fetchurl (ie: pre-stdenv) support HTTPS URLs?

**Comment by ryantm** [comment]
We can just close all of @r-ryantm's PRs and re-make them.

**Comment by Infinisil** [comment]
`nix-build -A stdenv --option substitute false --check` built without problems. Not sure if that actually recursively does the build though, I think it does.

I resolved the conflicts as well now.

**Comment by xeji** [comment]
Let's merge before new conflicts come up, and take it from there.

**Comment by Infinisil** [comment]
Alright, nice :D

**Pull request merged**